content/en/docs/tasks/administer-cluster/memory-manager.md

```diff
@@ -0,0 +1,390 @@
+---
+title: Memory Manager
+
+reviewers:
+- klueska
+- derekwaynecarr
+- TBD
+
+content_type: task
+min-kubernetes-server-version: v1.21
+---
+
+<!-- overview -->
+
+{{< feature-state state="alpha" for_k8s_version="v1.21" >}}
+
+The *Memory Manager* enables the feature of guaranteed memory (and hugepages) allocation for pods in the Guaranteed [QoS class](/docs/tasks/configure-pod-container/quality-service-pod/#qos-classes). 
+
+The *Memory Manager* employs hint generation protocol to yield the most suitable NUMA affinity for a pod. The *Memory Manager* feeds the central manager (*Topology Manager*) with these affinity hints. Based on both the hints and *Topology Manager* policy, the pod is rejected or admitted to the node.
+
+Moreover, *Memory Manager* ensures that the memory which a pod requests is allocated from a minimum number of NUMA nodes.
+
+## {{% heading "prerequisites" %}}
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+<!-- steps -->
+
+## How Memory Manager Operates?
+
+The Memory Manager currently offers the guaranteed memory (and hugepages) allocation for Pods in Guaranteed QoS class. To immediately put the Memory Manager into operation follow the guidelines in section [Memory Manager configuration](#memory-manager-configuration)), and subsequently, prepare and deploy a `Guaranteed` pod as illustrated in section [Placing a Pod in the Guaranteed QoS class](placing-a-pod-in-the-guaranteed-qos-class).
+
+The Memory Manager is a Hint Provider, and it provides topology hints for the Topology Manager which then aligns the requested resources according to these topology hints. It also enforces `cgroups` (i.e. `cpuset.mems`) for pods. The complete flow diagram concerning pod admission and deployment process is illustrated in [Memory Manager KEP: Design Overview][4].
+
+During this process, the Memory Manager updates its internal counters stored in [Node Map and Memory Maps][2] to manage guaranteed memory allocation. 
+
+The Memory Manager updates the Node Map during the startup and runtime as follows.

```

Comment:
This feels more like the introduction to a concept page, eg one that might live in https://kubernetes.io/docs/concepts/scheduling-eviction/

How about adding two pages, a concept page and also an explanation of a likely task involving this new feature?

The task-like content seems to start at the heading “Placing a Pod in the Guaranteed QoS class”

Reply:
Good point, but we followed Topology Manager doc structure, which already contains "How it works" section: 
https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/#how-topology-manager-works

Where would you exactly place the concept page for MM?
I found potential locations:
1. https://kubernetes.io/docs/concepts/extend-kubernetes/
Next to "Device Plugins"?
2. blog post, similarly to CPU Manager post: https://kubernetes.io/blog/2018/07/24/feature-highlight-cpu-manager/

Regarding "Concepts" branch in docs, Topology Manager is referenced here:
https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/ 
> "Once a Pod is assigned to a Node, the kubelet runs the Pod and allocates node-local resources. The topology manager can take part in node-level resource allocation decisions."

The reference points to: https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/

What we should do for sure, is to update Topology Manager documentation (https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/) as it is outdated:
e.g.: "Memory or Hugepages are not considered by the Topology Manager for NUMA alignment."

Reply:
I think I'd put the concept part in https://kubernetes.io/docs/concepts/policy/ to mention that you can configure a policy for setting topology hints.

There isn't yet a concept page for topology management; there should be. If there were, memory manager could be a section in the topology management concept page.

The concept page does not have to be particularly long or detailed. It's enough to outline the concept of topology management and hyperlink to other pages.